Mod poly and mpn_mod random generation (#2448)
flintlib/flint: doc/source/nmod_poly.rst

Randomization
--------------------------------------------------------------------------------

.. function:: void nmod_poly_rand(nmod_poly_t poly, flint_rand_t state, slong len)

    Generates a random polynomial with length up to ``len``.

.. function:: void nmod_poly_rand_monic(nmod_poly_t poly, flint_rand_t state, slong len)

    Generates a random monic polynomial with length up to ``len``.

.. function:: void nmod_poly_rand_irreducible(nmod_poly_t poly, flint_rand_t state, slong len)

    Generates a random irreducible polynomial with length up to ``len``.

.. function:: void nmod_poly_randtest(nmod_poly_t poly, flint_rand_t state, slong len)

>     Generates a random and sparse with increased probability polynomial with length up to ``len``.

Reviewer: ```suggestion
    Generates a random polynomial with length up to ``len``, with for each coefficient the probability of some special values increased (see :func:`n_randtest`). This function is intended for use in test code.
```

Reviewer: This is sparse with increased proba, but also special values are more likely than with a uniform distribution. So maybe, 
```suggestion
    Generates a random polynomial with length up to ``len``, sparse with increased probability, and with coefficients generated by :func:`n_randtest` .
```

Randomization
--------------------------------------------------------------------------------

.. function:: void nmod_poly_rand(nmod_poly_t poly, flint_rand_t state, slong len)

>     Generates a random polynomial with length up to ``len``.

Reviewer: ```suggestion
    Generates a random polynomial with length up to ``len`` and coefficients generated uniformly.
```

.. function:: void nmod_poly_rand_monic(nmod_poly_t poly, flint_rand_t state, slong len)

>     Generates a random monic polynomial with length up to ``len``.

Reviewer: ```suggestion
    Generates a random monic polynomial of length ``len`` and non-leading coefficients generated uniformly.
```

.. function:: void nmod_poly_rand_irreducible(nmod_poly_t poly, flint_rand_t state, slong len)

>     Generates a random irreducible polynomial with length up to ``len``.

Reviewer: ```suggestion
    Generates a random irreducible polynomial with length up to ``len`` with coefficients generated uniformly.
```

.. function:: void nmod_poly_randtest(nmod_poly_t poly, flint_rand_t state, slong len)

>     Generates a random and sparse with increased probability polynomial with length up to ``len``.

Author: ```suggestion
    Generates a random polynomial with length up to ``len``, sparse with increased probability, and with coefficients generated by :func:`n_randtest`. This function is intended for use in test code.
```

.. function:: void nmod_poly_randtest_monic(nmod_poly_t poly, flint_rand_t state, slong len)

>     Generates a random and sparse with increased probability monic polynomial with length ``len``.

Reviewer: See the comment above on a similar function.

>     Generates a random and sparse with increased probability monic polynomial with length ``len``.

Author: ```suggestion
    Generates a random monic polynomial with length ``len``. For each coefficient the probability of some special values is increased (see :func:`n_randtest`). In particular, the polynomial is sparse with increased probability. This function is intended for use in test code.
```
